tpm2tools/keys.go

```diff
@@ -205,30 +216,51 @@ func sealHelper(rw io.ReadWriter, parentHandle tpmutil.Handle, auth []byte, sens
 	} else {
 		inPublic.Attributes |= tpm2.FlagAdminWithPolicy
 	}
-	priv, pub, _, _, _, err := tpm2.CreateKeyWithSensitive(rw, parentHandle, tpm2.PCRSelection{}, "", "", inPublic, sensitive)
+
+	priv, pub, creationData, _, ticket, err := tpm2.CreateKeyWithSensitive(rw, parentHandle, certifyPCRsSel, "", "", inPublic, sensitive)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create key: %v", err)
+	}
+	certifiedPCR, err := ReadPCRs(rw, certifyPCRsSel)
 	if err != nil {
-		return nil, fmt.Errorf("failed to seal data: %v", err)
+		return nil, fmt.Errorf("Failed to read PCRs: %v", err)
+	}
+	computedDigest := computePCRDigest(certifiedPCR)
+
+	decodedCreationData, err := tpm2.DecodeCreationData(creationData)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to decode creation data: %v", err)
+	}
+
+	// make sure PCRs haven't being altered after sealing
+	if subtle.ConstantTimeCompare(computedDigest, decodedCreationData.PCRDigest) == 0 {
+		return nil, fmt.Errorf("PCRs have been modified after sealing")
 	}
 
 	sb := proto.SealedBytes{}
+	sb.CertifiedPcrs = certifiedPCR
 	sb.Priv = priv
 	sb.Pub = pub
+	sb.CreationData = creationData
+
+	t, err := tpmutil.Pack(ticket)
+	if err != nil {
+		return nil, err
+	}
+	sb.Ticket = t
+
 	return &sb, nil
 }
 
 // Unseal takes a private/public pair of buffers and attempts to reverse the
 // sealing process under the owner hierarchy using the SHA256 versions of the
 // provided PCRs.
+// CertificationOpt can be nil, which means no certify will be performed.
 // The Key k is used as the parent key.
```

Comment:
This comment needs an update to be consistent with `Seal()`. 

```go
// Unseal attempts to reverse the process of Seal(), using the PCRs, public, and
// private data in proto.SealedBytes. Optionally, a CertificationOpt can be
// passed, to verify the state of the TPM when the data was sealed. A nil value
// can be passed to skip certification.
```

Reply:
done